### PR TITLE
feat: pokevision overlay

### LINKED ISSUE #212: Add pokemon map support!!
please see github: Pokemon-Go-Map or PokemonVision.com, it could show up the nearby pokemon and what it exactly is, if pokemongo-webspoof can do this, that will be so so so so sos so so so COOL!

### PR DESCRIPTION
**Pull request by Mardaneus86**
Show an overlay of pokemons near you. Refreshes every 5 seconds and scans every minute. Not perfect yet, so please edit or suggest wherever appropriate.

It currently clears the mobx map every 30 seconds to prevent memory issues. Don't know if there is a better alternative to do this. Pokevision API provides unique id's per pokemon spawn that's found.

I have included all pokemons that are in the pokedex, like is supplied with Pokevisions website.

PS: not really sure if pokevision allows this kind of integration. Couldn't find anything about it on their website.

fixes #212

### CODE REVIEW

File: src/models/pokemon-location.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  isLoading.set(true);
+  console.log('checking spawns')
+  const url = `https://pokevision.com/map/data/${latitude}/${longitude}`
+
+  try {
+    const {data, status} = await axios.get(url);
+    if (status == 200 && data.status === "success") {
+      if (counter > 5) {
+        counter = 0;
+        spawns.clear();
```
Comment (Mardaneus86) [inline src/models/pokemon-location.js]: Don't really like this, there should be a better, optimized way to refresh the mobx map.

File: src/models/user-location.js [hunk 2]
```diff
@@ -40,6 +41,9 @@ const updateXcodeLocation = throttle(([ lat, lng ]) => {
   // track location changes for total distance & average speed
   stats.pushMove(lat, lng)
 
+  pokemonSpawns.checkPokemonSpawns(lat, lng)
```
Comment (Mardaneus86) [inline src/models/user-location.js]: Don't know if this is really the best location to do this.

File: src/views/map/pokespawn.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import React from 'react'
+import { observer } from 'mobx-react'
+import moment from 'moment'
+import pokedex from '../../models/pokedex.js'
+
+const PokemonSpawn = React.createClass({
+  getInitialState: function() {
+    return {
+    	expiration_time: this.props.spawn.expiration_time * 1000,
+    	endOfLife: {
```
Comment (Mardaneus86) [inline src/views/map/pokespawn.js]: Is there a better way to get a reactive countdown? This works but probably isn't the best way to solve this issue.

**Comment by 93732** [comment]
how do i compile this? or do i have to run from terminal?

**Comment by Mardaneus86** [comment]
Please read the Develop section of the readme for instructions.

**Comment by KoetseJ** [comment]
Maybe integration of the data from https://github.com/AHAAAAAAA/PokemonGo-Map would be better?

**Comment by bigreja** [comment]
PokemonGo-Map is very good and accurate, i've been using it, it would be interesting to see this integration instead.

**Comment by Mardaneus86** [comment]
Not really into python so I don't quite understand the way PokemonGo-Map is working internally. If someone has suggestions or wants to help on this feature please don't hesitate. 
I guess there aren't that many differences between PokemonGo-Map and Pokevision though? Or am I missing something? If all is well, the only file that would be needed to change is models/pokemon-location.js

**Comment by bigreja** [comment]
Mardaneus86 PokemonGo-Map runs on your computer, and uses Niantic API directly.

**Comment by Mardaneus86** [comment]
I guess we would need a JS port for that API. I think adding python as a dependency wouldn't be the correct thing to do. Does anyone know if there are JS implementations for the pgoapi?

@iam4x : what do you think?

**Comment by iam4x** [comment]
First of all, thank's a lot for the submission @Mardaneus86 👍 

If we want to run our own pokemongo-map project we have to make integrate a JS solution, will take much more time than use pokevision which is straightforward as implementation.

I think we want things to work before having perfect solutions, so let's integrate this pokevision before doing something else.

**Comment by KoetseJ** [comment]
How about adding a option for users to supply there own PokemongGo-Map? You currently can 1 click launce a free online docker instace that will only require a pokemon go account.

Then you can switch between pokevision, and a PokemonGo-Map server?

The data that will be supplied, will look like this:
https://freepokemonheroku.herokuapp.com/raw_data?pokemon=true&pokestops=true&gyms=true&scanned=true

Updating your location would require the following post call:
https://freepokemonheroku.herokuapp.com/next_loc?lat=51.9244201&lon=4.477732500000002

You would only need to update the location once you move 1k away from the last location.

**Comment by iam4x** [comment]
I've tested the branch, there's really bad performance with all these pokémons on screen. I have to find what's wrong with that.

Without that the feature is really good 👍 

@KoetseJ I understand your thoughts, but will be faster to integrate this solution at first. I will be open to new PR with PokemonGo-Map project ;)

**Comment by Mardaneus86** [comment]
@iam4x I am aware of the performance issues if it's a crowded area. No idea how to fix it at this moment though. Let me know if you happen to know what's wrong.

**Comment by Mardaneus86** [comment]
Pokevision just implemented some sort of browser check to prevent scraping. Will try to figure out a workaround. If that's not easily possible, we should probably go for the PokemonGo-Map approach instead.

**Comment by h4v1nfun** [comment]
@Mardaneus86 They are saying that this still work after cloudfare -> https://github.com/dgoguerra/pokego-scan

**Comment by iam4x** [comment]
It's still working after cloudfare, somehow they enabled cloudfare yesterday for some time then disabled it   Oo

**Comment by iam4x** [comment]
@Mardaneus86 I rewrote the feature in a more reactive way, I don't have performance issues. Thank's a lot again for the PR I took the most of it with a big refactoring 👍 

💘

**Pull request closed without merging**